apps/frontend/src/pages/meeting-details/meeting-details.tsx

```diff
@@ -1,21 +1,196 @@
+		defaultValues: DEFAULT_SEARCH_VALUE,
+		validationSchema: searchInputValidationSchema,
+	});
+
+	useEffect(() => {
+		const sharedToken = searchParameters.get("token");
+		void dispatch(
+			meetingDetailsActions.getMeetingDetailsById({
+				id: Number(id),
+				sharedToken: sharedToken ?? undefined,
+			}),
+		);
+	}, [id, dispatch, searchParameters]);
+
+	const handleTranscriptionSearch = useCallback((value: string) => {
+		// TODO: implement handleTranscriptionSearch logic
+		return value;
+	}, []);
+
+	const handleShareClick = useCallback(() => {
+		if (!meeting || !meeting.id) {
+			notification.error(NotificationMessage.MEETING_DATA_IS_NOT_AVAILABLE);
+
+			return;
+		}
+
+		const shareMeeting = async (): Promise<void> => {
+			try {
+				const { publicUrl } = await meetingDetailsApi.getPublicShareUrl(
+					meeting.id,
+				);
+				const host = import.meta.env["VITE_APP_HOST"] as string;
+				void navigator.clipboard.writeText(`${host}${publicUrl}`);
+				notification.success(NotificationMessage.PUBLIC_LINK_COPIED_SUCCESS);
+			} catch (error: unknown) {
+				notification.error(NotificationMessage.SHARE_LINK_GENERATION_FAILED);
 
-	const stopRecording = useCallback(() => {
-		void dispatch(meetingActions.stopRecording({ id: id as string }));
-	}, [dispatch, id]);
+				throw error;
+			}
+		};
+
+		void shareMeeting();
+	}, [meeting]);
+
+	if (!id || dataStatus === DataStatus.REJECTED) {
+		return <Navigate replace to={AppRoute.NOT_FOUND} />;
+	}
+
+	if (dataStatus === DataStatus.PENDING) {
+		return <Loader hasOverlay isLoading />;
+	}
+
+	if (!meeting) {
+		return (
+			<div className={styles["meeting-details__not-found"]}>
+				Meeting not found.
+			</div>
+		);
+	}
 
 	return (
 		<>
-			<div>Meeting Nº {id}</div>
-			<Button label="Stop Recording" onClick={stopRecording} />
-			<div>
-				<PlayerTrack audioUrl="https://audio-samples.github.io/samples/mp3/wavenet_unconditional/voxceleb2/sample-5.mp3" />
+			<div className={styles["meeting-details"]}>
+				<div className={styles["meeting-details__header"]}>
+					<h1 className={styles["meeting-details__title"]}>
+						Meeting #{meeting.id} |{" "}
+						{formatDate(new Date(meeting.createdAt), "D MMMM hA")}
+					</h1>
+					<div className={styles["meeting-details__actions"]}>
+						{user && (
+							<button
+								className={styles["action-button"]}
+								onClick={handleShareClick}
+							>
+								<Icon className={styles["action-button__share"]} name="share" />
+							</button>
+						)}
+						<Button label="Export" />
+					</div>
+				</div>
+
+				<div className={styles["meeting-details__content"]}>
+					<div className={styles["meeting-details__transcription-panel"]}>
+						<div className={styles["panel-header"]}>
+							<h3 className={styles["panel-header__text"]}>TRANSCRIPT</h3>
+							<div className={styles["panel-header__search"]}>
+								<SearchInput
+									control={control}
+									errors={errors}
+									hasVisuallyHiddenLabel={true}
```

Comment:
Should work like this
```suggestion
									hasVisuallyHiddenLabel
```